DFC.ServiceTaxonomy.ContentApproval/Controllers/ContentApprovalController.cs

```diff
@@ -36,11 +40,86 @@ public async Task<IActionResult> RequestApproval(string contentItemId, string re
                 return Forbid();
             }
 
-            contentItem.Alter<ContentApprovalPart>(p => p.ApprovalStatus = ContentApprovalStatus.ReadyForReview_ContentDesign);
+            contentItem.Alter<ContentApprovalPart>(p => p.ReviewStatus = ContentReviewStatus.ReadyForReview);
             await _contentManager.SaveDraftAsync(contentItem);
-            _notifier.Success(H[$"{0} is now ready to be reviewed.", contentItem.DisplayText]);
+            _notifier.Success(H["{0} is now ready to be reviewed.", contentItem.DisplayText]);
 
             return Redirect(returnUrl);
         }
+
+        public async Task<IActionResult> Review(string contentItemId, string returnUrl)
+        {
+            var contentItem = await _contentManager.GetAsync(contentItemId, VersionOptions.Latest);
+            if (contentItem == null)
+            {
+                return NotFound();
+            }
+            if (!await _authorizationService.AuthorizeAsync(User, Permissions.CanPerformReviewPermissions.CanPerformReviewPermission, contentItem))
+            {
+                return Forbid();
+            }
+
+            var contentApprovalPart = contentItem.As<ContentApprovalPart>();
+            var reviewStatus = contentApprovalPart.ReviewStatus;
+            if (reviewStatus == ContentReviewStatus.NotInReview)
+            {
+                _notifier.Warning(H["This item is currently in the '{0}' state and therefore is no longer available for review", contentApprovalPart.IsPublished ? "Published" : "In Draft"]);
+                return Redirect(returnUrl);
+            }
+
+            if (reviewStatus == ContentReviewStatus.ReadyForReview)
```

Comment:
Could we use a switch statement, or else if's here, so that we don't unnecessarily keep checking the status?

Reply:
Agree, have changed to an 'else if'